dist/gl-matrix.js

```diff
@@ -5725,31 +5826,78 @@ THE SOFTWARE.
     return out;
   }
   /**
-   * Creates a quaternion from the given euler angle x, y, z.
+   * Creates a quaternion from the given euler angle x, y, z using the provided intrinsic order for the conversion.
    *
    * @param {quat} out the receiving quaternion
    * @param {x} Angle to rotate around X axis in degrees.
    * @param {y} Angle to rotate around Y axis in degrees.
    * @param {z} Angle to rotate around Z axis in degrees.
+   * @param {order} Intrinsic order for conversion, default is ZYX.
    * @returns {quat} out
    * @function
    */
 
   function fromEuler(out, x, y, z) {
+    var order = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : 'zyx';
     var halfToRad = 0.5 * Math.PI / 180.0;
     x *= halfToRad;
-    y *= halfToRad;
     z *= halfToRad;
+    y *= halfToRad;
     var sx = Math.sin(x);
     var cx = Math.cos(x);
     var sy = Math.sin(y);
     var cy = Math.cos(y);
     var sz = Math.sin(z);
     var cz = Math.cos(z);
-    out[0] = sx * cy * cz - cx * sy * sz;
-    out[1] = cx * sy * cz + sx * cy * sz;
-    out[2] = cx * cy * sz - sx * sy * cz;
-    out[3] = cx * cy * cz + sx * sy * sz;
+
+    if (typeof order != 'string') {
+      order = 'zyx';
+    }
+
+    switch (order.toLowerCase()) {
+      case 'xyz':
+        out[0] = sx * cy * cz + cx * sy * sz;
+        out[1] = cx * sy * cz - sx * cy * sz;
+        out[2] = cx * cy * sz + sx * sy * cz;
+        out[3] = cx * cy * cz - sx * sy * sz;
+        break;
+
+      case 'xzy':
+        out[0] = sx * cy * cz - cx * sy * sz;
+        out[1] = cx * sy * cz - sx * cy * sz;
+        out[2] = cx * cy * sz + sx * sy * cz;
+        out[3] = cx * cy * cz + sx * sy * sz;
+        break;
+
+      case 'yxz':
+        out[0] = sx * cy * cz + cx * sy * sz;
+        out[1] = cx * sy * cz - sx * cy * sz;
+        out[2] = cx * cy * sz - sx * sy * cz;
+        out[3] = cx * cy * cz + sx * sy * sz;
+        break;
+
+      case 'yzx':
+        out[0] = sx * cy * cz + cx * sy * sz;
+        out[1] = cx * sy * cz + sx * cy * sz;
+        out[2] = cx * cy * sz - sx * sy * cz;
+        out[3] = cx * cy * cz - sx * sy * sz;
+        break;
+
+      case 'zxy':
+        out[0] = sx * cy * cz - cx * sy * sz;
+        out[1] = cx * sy * cz + sx * cy * sz;
+        out[2] = cx * cy * sz + sx * sy * cz;
+        out[3] = cx * cy * cz - sx * sy * sz;
+        break;
+
+      case 'zyx':
+      default:
```

Comment:
Falling back to `zyx` when specifying an order that isn't supported (e.g. `xyx`) feels like an API footgun, I would rather throw an error here.